### ISSUE TITLE
Scaling!

### ISSUE THREAD

**Issue by sfinexer** [vlachoudis/bCNC#978]
I suggest to add a scaling option. That it was possible to change in proportion the object size (X-Y) or (X) or (Y) or (Z) or (X-Y-Z)

It is very useful when there is a certain initial object!!! and it is necessary to make similar but other size!

**Comment by MARIOBASZ** [comment]
First images of the scalar plugin
Original circle:
![image](https://user-images.githubusercontent.com/11743403/46108594-cbaa2680-c1b4-11e8-99aa-8a073594539f.png)
Scaling 2:1 in xy plane
![image](https://user-images.githubusercontent.com/11743403/46108675-fe541f00-c1b4-11e8-8bd7-59bb10b484a7.png)

2: 1 scale in X direction; scale 1.5: 1 in Y direction
![image](https://user-images.githubusercontent.com/11743403/46108799-46734180-c1b5-11e8-932f-1e0023026c41.png)

Notes:
1-They are all straight sections, there are no curved lines.
2-The idea was to make a plugin for backlash problems, and, using the same concept, modify it to make a plugin to scale. I have reversed the steps, with a small modification I will upload a plugin for backlash (maybe I will also add it to correct orthogonality problems)

**Comment by jpagliaccio** [comment]
I love the scaling - it's very useful. In version 0.9.14-dev, the new scaled path is all G0 rapid. Am I doing something wrong? My scale settings are X Scale = 2, Y Scale = 2, Z Scale = 1.

**Comment by MARIOBASZ** [comment]
I'm glad you find it useful!
I do not remember modifying something, I will review it.
Maybe the depth z is equal to or greater

![F8D05BA2-CAD1-45E0-A0B1-77D3A71924C3](https://user-images.githubusercontent.com/11743403/70148160-64bfb180-1684-11ea-87fa-66bc91355952.jpeg)
 than zero? (I am very bad in memory, I do not remember well as I did)
In my machine I have made another pado, from a circle I came to this:
